From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rai san <admin@raic.tech>
Date: Wed, 15 May 2024 18:23:55 +0900
Subject: [PATCH] feat: Rales (beta) v2


diff --git a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
index 462a6eed350fd660ddaf25d567bb6e97b77d0b2b..2c10cd3231203194d2abc5d0f2df232019e33e7d 100644
--- a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
+++ b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
@@ -20,7 +20,7 @@ import java.util.stream.StreamSupport;
 public class PaperVersionFetcher implements VersionFetcher {
     private static final java.util.regex.Pattern VER_PATTERN = java.util.regex.Pattern.compile("^([0-9\\.]*)\\-.*R"); // R is an anchor, will always give '-R' at end
     // Purpur start
-    private static final String DOWNLOAD_PAGE = "https://purpurmc.org/downloads";
+    private static final String DOWNLOAD_PAGE = "https://rales.raic.tech"; // Rales
     private static int distance = -2; public int distance() { return distance; }
     // Purpur end
     private static @Nullable String mcVer;
@@ -33,8 +33,8 @@ public class PaperVersionFetcher implements VersionFetcher {
     @Nonnull
     @Override
     public Component getVersionMessage(@Nonnull String serverVersion) {
-        String[] parts = serverVersion.substring("git-Purpur-".length()).split("[-\\s]"); // Purpur
-        final Component updateMessage = getUpdateStatusMessage("PurpurMC/Purpur", "ver/" + getMinecraftVersion(), parts[0]); // Purpur
+        String[] parts = serverVersion.substring("git-Rales-".length()).split("[-\\s]"); // Purpur
+        final Component updateMessage = getUpdateStatusMessage("UpLauncher/Rales", "ver/" + getMinecraftVersion(), parts[0]); // Purpur
         final Component history = getHistory();
 
         return history != null ? Component.join(net.kyori.adventure.text.JoinConfiguration.separator(Component.newline()), history, updateMessage) : updateMessage; // Purpur
@@ -68,17 +68,17 @@ public class PaperVersionFetcher implements VersionFetcher {
 
         switch (distance) {
             case -1:
-                return Component.text("* Error obtaining version information", NamedTextColor.RED); // Purpur
+                return Component.text("* バージョン情報の取得中にエラーが発生しました。", NamedTextColor.RED); // Purpur
             case 0:
-                return Component.text("* You are running the latest version", NamedTextColor.GREEN); // Purpur
+                return Component.text("* 最新バージョンで実行しています。", NamedTextColor.GREEN); // Purpur
             case -2:
-                return Component.text("* Unknown version", NamedTextColor.YELLOW); // Purpur
+                return Component.text("* 不明なバージョン", NamedTextColor.YELLOW); // Purpur
             default:
-                return Component.text("* You are " + distance + " version(s) behind", NamedTextColor.YELLOW) // Purpur
+                return Component.text("* Rales は、" + distance + " バージョン古いです。", NamedTextColor.YELLOW) // Purpur
                         .append(Component.newline())
-                        .append(Component.text("Download the new version at: ")
+                        .append(Component.text("新しいバージョンをダウンロード: ")
                                 .append(Component.text(DOWNLOAD_PAGE, NamedTextColor.GOLD)
-                                        .hoverEvent(Component.text("Click to open", NamedTextColor.WHITE))
+                                        .hoverEvent(Component.text("クリックして開く", NamedTextColor.WHITE))
                                         .clickEvent(ClickEvent.openUrl(DOWNLOAD_PAGE))));
         }
     }
