From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rai san <admin@raic.tech>
Date: Wed, 15 May 2024 20:16:21 +0900
Subject: [PATCH] feat: Default Create rales.yml


diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 6d8e9cf4ee496b8748654e792ce438f512c3d946..2bf9d5b48575ddc60d8c5ca86dda7eb9ef3c5369 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -184,6 +184,14 @@ public class Main {
                     .describedAs("Yml file");
                 // Purpur end
 
+                // Rales start
+                acceptsAll(asList("rales", "rales-settings"), "File for rales settings")
+                    .withRequiredArg()
+                    .ofType(File.class)
+                    .defaultsTo(new File("rales.yml"))
+                    .describedAs("Yml file");
+                // Rales end
+
                 // Paper start
                 acceptsAll(asList("server-name"), "Name of the server")
                         .withRequiredArg()
