From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rai san <admin@raic.tech>
Date: Wed, 15 May 2024 20:16:21 +0900
Subject: [PATCH] feat: Default Create rales.yml


diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index a0a7bfe8f5a40b945b83096de1efd650adeb9ee3..d73173b97a40ce9f6a4b6a2d0c1e636d956a27da 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -183,6 +183,14 @@ public class Main {
                     .describedAs("Yml file");
                 // Purpur end
 
+                // Rales start
+                acceptsAll(asList("rales", "rales-settings"), "File for rales settings")
+                    .withRequiredArg()
+                    .ofType(File.class)
+                    .defaultsTo(new File("rales.yml"))
+                    .describedAs("Yml file");
+                // Rales end
+
                 // Paper start
                 acceptsAll(asList("server-name"), "Name of the server")
                         .withRequiredArg()
