From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rai san <admin@raic.tech>
Date: Fri, 17 May 2024 16:57:23 +0900
Subject: [PATCH] fix: Rales Updating URL


diff --git a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
index a51d142afc631cd09c8f3b8a6962b8b311687060..bd2f32336d8a3e7d145e6c600a4793a66d8b7591 100644
--- a/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
+++ b/src/main/java/com/destroystokyo/paper/PaperVersionFetcher.java
@@ -87,7 +87,7 @@ public class PaperVersionFetcher implements VersionFetcher {
         if (siteApiVersion == null) { return -1; }
         try {
             try (BufferedReader reader = Resources.asCharSource(
-                new URL("http://localhost:3030/update" + siteApiVersion), // Purpur // Rales
+                new URL("http://localhost:3030/update/" + siteApiVersion), // Purpur // Rales
                 Charsets.UTF_8
             ).openBufferedStream()) {
                 JsonObject json = new Gson().fromJson(reader, JsonObject.class);
